sdk/apimanagement/api-management-custom-widgets-scaffolder/src/templates/vue/src/main.app.ts

```diff
@@ -0,0 +1,18 @@
+import {createApp} from "vue"
+import {askForSecrets} from "@azure/api-management-custom-widgets-tools"
+import App from "./components/app/index.vue"
+
+const secretsPromise = askForSecrets("app")
+const requestPromise = new Promise(async (resolve) => {
+  const secrets = await secretsPromise
```

Comment:
Is this promise nesting a common pattern in vue? It seems a bit convoluted to me, can we just await for the secrets promise and then fetch without wrapping in into another promise?

```ts
const secrets = await askForSecrets("app");
const request = (url: string) => fetch(
    `${secrets.managementApiUrl}${url}?api-version=${secrets.apiVersion}`,
    secrets.token ? {headers: {Authorization: secrets.token}} : undefined,
  ))
```

Reply:
This would not work as then those functions
```
const app = createApp(App)
app.mount("#root")
```
would not run immediately, which we don't want